m2m-transfers/m2m-transfers-json-contract.md

```diff
@@ -0,0 +1,196 @@
+#JSON contract for BANNO.M2MTRANSFERS.V1.POW
+
+## GET PRELOADDATA STATE
+### Client Request (PRELOADDATA)
+
+```json
+{
+  "rgState": "PRELOADDATA",
+  "powerOnFileName": "BANNO.M2MTRANSFERS.V1.POW",
+  "userChrList": [
+    { "id": 1, "value": "1234567890" }, // 10-digit account
+    { "id": 2, "value": "" }, // userchar[2-5] is unused
+    { "id": 3, "value": "" },
+    { "id": 4, "value": "" },
+    { "id": 5, "value": "" }
+  ],
+  "userNumList": [], //  usernum[1-5] is unused
+  "rgSession": 1
+}
+```
+
+### PowerOn Successful Response (PRELOADDATA)
+list of institution and member limits, eligible shares, list of scheduled transfers, list of saved recipients
+
+```json
+{
+	"currentState": {
+		"enforceLimits": true,
+		"transferLimits": [{
+			"institutionCount": "5",
+			"institutionAmount": "10000.00",
+			"memberLimitCount": "3",
+			"memberLimitAmount": "5000.00",
+			"memberActualCount": "1",
+			"memberActualAmount": "500.00"
+		}],
+		"availableShares": [{
+				"transferSLId": "0000800005S0002",
+				"transferName": "SUMMER SAVER",
+				"available": "50.00"
+			},
+			{
+				"transferSLId": "0000800005S0010",
+				"transferName": "REGULAR CHECKING",
+				"available": "500.00"
+			},
+			{
+				"transferSLId": "0000800005S0015",
+				"transferName": "ULTIMATE CHECKING (10)",
+				"available": "0.00"
+			}
+		],
+
+		"scheduledTransfers": [{
+				"trnsferLoc": "215",
+				"sourceAccount": "1234567890S0020",
+				"accountName": "CHECKING",
+				"transferAmt": "120.00",
+				"recipientName": "Emily Fitch",
+				"recipientMemberId": "9876543210",
+				"recipientAccount": "[9876543210L0001,savings,checking]",
+				"recipientNickname": "Emmy",
+				"startDate":"07/31/21",
+				"endDate":"12/31/2021",
+				"transferFrequency": "weekly",
+				"day1": "",
+				"day2": "",
+				"saveRecipient": false,
+				"memo": "Internal Note created by member"
+			},
+			{
+				"transferLoc": "395",
+				"sourceAccount": "1234567890S0001",
+				"accountName": "SAVINGS",
+				"transferAmt": "120.00",
+				"recipientName": "Howard Cleo",
+				"recipientMemberId": "9876543210",
+				"recipientShareId": "9876543210L0001",
+				"recipientNickname": "Howie",
+				"nextTransferDate": "12/15/2020",
+				"startDate":"07/03/21",
+				"endDate":"12/31/2025",
+				"transferFrequency": "semi-monthly",
+				"day1": "3",
+				"day2": "31",
+				"saveRecipient": false,
+				"memo": "Internal Note created by member"
+			}
+		],
+		"savedRedipients": [{
+				"recipientLoc": "5443231543",
```

Comment:
what is `recipientLoc`?

Reply:
Its the Locator code (unique serial number) assigned to a record - in this case, the external account record - at the time it's created. It allows the program to definitively target the correct record when updating or deleting it.

Reply:
Ok thanks, that makes sense!